base/statistics.jl

```diff
@@ -177,29 +241,48 @@ _median(v::AbstractArray{T}, ::Colon) where {T} = median!(copyto!(Array{T,1}(und
 # for now, use the R/S definition of quantile; may want variants later
 # see ?quantile in R -- this is type 7
 """
-    quantile!([q, ] v, p; sorted=false)
+    quantile!([q::AbstractArray, ] v::AbstractVector, p; sorted=false)
 
-Compute the quantile(s) of a vector `v` at the probability or probabilities `p`, which
-can be given as a single value, a vector, or a tuple. If `p` is a vector, an optional
+Compute the quantile(s) of vector `v` at a specified probability or vector or tuple of
+probabilities `p` on the interval [0,1]. If `p` is a vector, an optional
 output array `q` may also be specified. (If not provided, a new output array is created.)
 The keyword argument `sorted` indicates whether `v` can be assumed to be sorted; if
-`false` (the default), then the elements of `v` may be partially sorted.
-
-The elements of `p` should be on the interval [0,1], and `v` should not have any `NaN`
-values.
+`false` (the default), then the elements of `v` will be partially sorted in-place.
 
 Quantiles are computed via linear interpolation between the points `((k-1)/(n-1), v[k])`,
 for `k = 1:n` where `n = length(v)`. This corresponds to Definition 7 of Hyndman and Fan
 (1996), and is the same as the R default.
 
 !!! note
-    Julia does not ignore `NaN` values in the computation: `quantile!` will
-    throw an `ArgumentError` in the presence of `NaN` values in the data array.
-    Use the [`missing`](@ref) type to represent missing values, and the
-    [`skipmissing`](@ref) function to omit them.
+    An `ArgumentError` is thrown if `v` contains `NaN` or [`missing`](@ref) values.
```

Comment:
I have preserved the current behavior, which is inconsistent with `median` (and `mean`, `sum`, etc.). I can change it now to return `missing` instead of throwing an error, but we can do that at any time in the 1.x series since that's not breaking.

Reply:
It seems unfortunate if `median` and `quantile` don't have the same behavior wrt `missing`s

Reply:
I'm fine with changing `quantile` to return `missing` instead of throwing an error, but that requires a bit more work/thought given that `quantile` returns an array when passed several quantile levels, and yet we probably don't want to make it an `Array{Union{T,Missing}}`. So I'd rather change this in another PR so that we can discuss it more thoroughly.

Reply:
Fine with me

Reply:
Filed https://github.com/JuliaLang/julia/issues/27905 so that we don't forget.